Array_Control/Array_Control.ino

```diff
@@ -148,48 +136,46 @@ battery batts_array[NUM_BATTS]; // {batt_0, batt_1, ...}
+        for(int i = 0; i < NUM_BATTS; i++){
 
-    //Fet assignment array. The index of the array is the battery case. ie, [0][0] is case 0, [1][0] is case 1
-    int FET_assignments[NUM_BATTS][NUM_FETS] = { 
-        { OUT_FET12, OUT_FET15, OUT_FET17, OUT_FET19, OUT_FET21 }, 
-        { OUT_FET11, OUT_FET14, OUT_FET17, OUT_FET19, OUT_FET21 },
-        { OUT_FET11, OUT_FET13, OUT_FET16, OUT_FET19, OUT_FET21 },
-        { OUT_FET11, OUT_FET13, OUT_FET15, OUT_FET18, OUT_FET21 },
-        { OUT_FET11, OUT_FET13, OUT_FET15, OUT_FET17, OUT_FET20 } };
-
-    //iterate over the array to perform pin assignments and set charging flags low
-    for (int battery_case_index = 0; battery_case_index < NUM_BATTS; battery_case_index++)
-    {
-        //Since we are already iterating over the array, lets initialize the adc pin and the is_charging flag in the structure
-        batts_array[battery_case_index].adc_pin_assignment = adc_pins[battery_case_index];
-        Serial.print("Assigned ADC Pin: ");
-        Serial.println(batts_array[battery_case_index].adc_pin_assignment);
+            //printf("Battery %d configuration\n", i);
+            Serial.print("Battery "); Serial.print(i); Serial.println(" configuration");
 
-        batts_array[battery_case_index].is_charging        = false;
+            //printf("    is_charging: %d\n", batts_array[i].is_charging);
+            Serial.print("      is_charging: "); Serial.println(batts_array[i].is_charging);
 
-        //itertate in 2d to assign FET config to each battery
-        for (int FET_assignment_index = 0; FET_assignment_index < NUM_FETS ; FET_assignment_index++)
-        {
-            pinMode(FET_assignment_index, OUTPUT);  //might as well set the pins to outputs while we are iterating
-        
-            if (DEBUG)
-            {
-            Serial.print("Batt_Case ");
-            Serial.print(battery_case_index);
-            Serial.print(", OUT_FET ");
-            Serial.print(FET_assignment_index);
-            Serial.println(", configured to output");
+            //printf("    adc_pin_assignment: %d\n", batts_array[i].adc_pin_assignment);
+            Serial.print("      adc_pin_assignement: "); Serial.println(batts_array[i].adc_pin_assignment);
+
+            //printf("    OUT_FETS: ");
+            Serial.print("      OUT_FETS: ");
+            
+            for(int ii = 0; ii < 5; ii++){
+                //printf("%d, ", batts_array[i].OUT_FETS[ii]);
+                Serial.print(batts_array[i].OUT_FETS[ii]);
             }
-            //set the FETS array in the structure to the 2d element of the FET_assignments array
-            batts_array[battery_case_index].FETS[FET_assignment_index] = FET_assignments[battery_case_index][FET_assignment_index];
+            //printf("\n");
+            Serial.println(" ");
+
+            //printf("        CHARGE_FETS: ");
+            Serial.print("      CHARGE_FETS: ");
+
+            for(int ii = 0; ii < 2; ii++){
```

Comment:
I would change the condition "ii < 2" to "ii < NUM_CHARGE_FETS_PER_BATT" to keep it more general. The same applies above on line 163 for the conditional for output FETs.

Reply:
fixed in latest commit